lib/pure/asyncdispatch.nim

```diff
@@ -246,13 +256,18 @@ when defined(windows) or defined(nimdoc):
     PDispatcher* = ref object of PDispatcherBase
       ioPort: Handle
       handles: HashSet[AsyncFD]
+      virtualHandles: Table[VirtualFD, AsyncEvent] # pseudo handles for custom AsyncEvents.
+      nextVirtualHandle: VirtualFD
+      virtualMuxHandle: Handle     # all the virtual handles get multiplexed through a single real handle.
+      virtualPCD: PostCallbackDataPtr # needed for the selector callback plumbing.
```

Comment:
perhaps all these fields could be put in a separate object?

Reply:
Sure. I left it all together for simplicity (and I was organically adding fields as I needed them), but I can see the organizational advantages of making a separate object.